src/charm.py

```diff
@@ -19,6 +29,139 @@ class KratosCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+        self._container_name = "kratos"
+        self.container = self.unit.get_container(self._container_name)
+
+        self.resource_handler = KubernetesResourceHandler(
+            template_files=self._template_files,
+            context=self._context,
+            field_manager=self.model.app.name,
+        )
+        self.lightkube_client = Client(namespace=self.model.name, field_manager="lightkube")
+
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.kratos_pebble_ready, self._on_pebble_ready)
+        self.framework.observe(self.on.remove, self._on_remove)
+
+    @property
+    def _template_files(self):
+        src_dir = Path("src/manifests")
+        manifests = [file for file in glob.glob(f"{src_dir}/*.yaml")]
+        return manifests
+
+    @property
+    def _context(self):
+        context = {
+            "app_name": self.model.app.name,
+        }
+        return context
+
+    @property
+    def _pebble_layer(self) -> Layer:
+        pebble_layer = {
+            "summary": "kratos layer",
+            "description": "pebble config layer for kratos",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Kratos Operator layer",
+                    "startup": "enabled",
+                    "command": "kratos serve all --config /etc/config/kratos.yaml",
+                    "environment": {
+                        "DSN": "postgres://username:password@10.152.183.152:5432/postgres",
+                        "COURIER_SMTP_CONNECTION_URI": "smtps://test:test@mailslurper:1025/?skip_ssl_verify=true",
+                    },
+                }
+            },
+            "checks": {
+                "ready": {
+                    "override": "replace",
+                    "http": {"url": "http://localhost:4434/admin/health/ready"},
+                },
+                "alive": {
+                    "override": "replace",
+                    "http": {"url": "http://localhost:4434/admin/health/ready"},
```

Comment:
The `/health/alive` endpoint is for checking the http server status only, while `/health/ready` checks both the server and database connection. Maybe just the second one is enough? Anyway, they seem to be duplicated:
```suggestion
                    "http": {"url": "http://localhost:4434/admin/health/alive"},
```